tests/test_api.py

```diff
@@ -388,11 +403,12 @@ def test_component_detail_dev(client, api_path):
 @pytest.mark.django_db
 def test_component_write_not_allowed(client, api_path):
```

Comment:
Let's test what happens If the user authenticates. AKA does the PUT / POST succeed, or does it continue to fail with the same 405 error as below?
```suggestion
@pytest.mark.django_db
def test_component_write_not_allowed_if_authenticated(client, api_path):
    c = ComponentFactory(name="curl")

    # User for OpenLCS authentication
    olcs_user = User.objects.create_user(username="olcs", email="olcs@example.com")
    olcs_token = Token.objects.create(user=olcs_user, key="mysteries_quirewise_volitant_woolshed")
    
    # Authenticate
    client.credentials(HTTP_AUTHORIZATION=f"Token {olcs_token.key}")

    response = client.put(f"{api_path}/components/{c.uuid}", params={"name": "wget"})
    assert response.status_code == 405
    response = client.post(f"{api_path}/components", data={"name": "curl"})
    assert response.status_code == 405


@pytest.mark.django_db
def test_component_write_not_allowed_if_unauthenticated(client, api_path):
```